File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    stack = [0] * bridge_length
    sum = 0
    while len(truck_weights) != 0:
        answer += 1
        head = stack.pop(0)
        if head != 0:
            sum -= head
        if sum + truck_weights[0] <= weight:
            truck_start = truck_weights.pop(0)
            stack.append(truck_start)
            sum += truck_start
        else:
            stack.append(0)
    answer += bridge_length
    return answer

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.swift
import Foundation

func solution(_ bridgeLength: Int, _ weight: Int, _ truckWeights: [Int]) -> Int {
    var answer = 0
    var stack = Array(repeating: 0, count: bridgeLength)
    var sum = 0
    var truckWeights = truckWeights
    
    while !truckWeights.isEmpty {
        answer += 1
        let head = stack.removeFirst()
        if head != 0 {
            sum -= head
        }
        if sum + truckWeights[0] <= weight {
            let truckStart = truckWeights.removeFirst()
            stack.append(truckStart)
            sum += truckStart
        } else {
            stack.append(0)
        }
    }
    answer += bridgeLength
    return answer
}